src/Sextant.WinUI/NavigationView.xaml.cs

```diff
@@ -0,0 +1,398 @@
+
+                            mainFrame.Navigate(pageType, null, new SuppressNavigationTransitionInfo());
+                            if (mainFrame.Content is IViewFor viewForReset)
+                            {
+                                viewForReset.ViewModel = viewModel;
+                            }
+                            else
+                            {
+                                var contentTypeName = mainFrame.Content?.GetType().ToString() ?? "Unknown Type";
+                                _logger.Debug(
+                                    $"The view ({contentTypeName}) does not implement IViewFor<>.  Cannot set ViewModel of type, {viewModel.GetType()}, on view.");
+                            }
+
+                            mainFrame.BackStack.Clear();
+
+                            return Observable.Return(Unit.Default);
+                        }
+
+                        NavigationTransitionInfo animation;
+                        if (animate)
+                        {
+                            animation = new EntranceNavigationTransitionInfo();
+                        }
+                        else
+                        {
+                            animation = new SuppressNavigationTransitionInfo();
+                        }
+
+                        _mirroredPageStack.Push(viewModel);
+
+                        mainFrame.Navigate(pageType, null, animation);
+                        if (mainFrame.Content is IViewFor viewFor)
+                        {
+                            viewFor.ViewModel = viewModel;
+                        }
+                        else
+                        {
+                            var contentTypeName = mainFrame.Content?.GetType().ToString() ?? "Unknown Type";
+                            _logger.Debug(
+                                $"The view ({contentTypeName}) does not implement IViewFor<>.  Cannot set ViewModel of type, {viewModel.GetType()}, on view.");
+                        }
+
+                        return Observable.Return(Unit.Default);
+                    });
+
+        private static Type LocatePageTypeFor(object viewModel, string? contract)
+        {
+            var uwpViewTypeResolver = Locator.Current.GetService<ViewTypeResolver>(contract);
+
+            var viewType = uwpViewTypeResolver?.ResolveViewType(viewModel.GetType());
+
+            if (viewType is null)
+            {
+                throw new InvalidOperationException(
+                    $"No view could be located for type '{viewModel.GetType().FullName}', contract '{contract}'. Be sure Splat has an appropriate registration.");
+            }
+
+            return viewType;
+        }
+
+        private Page LocatePageFor(object viewModel, string? contract)
+        {
+            var view = _viewLocator.ResolveView(viewModel, contract);
+            var page = view as Page;
+
+            if (view is null)
```

Comment:
The UWP-version also has an duplicated `view is null` check:
```
            if (view is null)
            {
                throw new InvalidOperationException($"Cannot find view for '{viewModel.GetType().FullName}', contract '{contract}' does not implement IViewFor.");
            }
```
this is dead code.
What should it check?
I have removed it.